emcc.py

```diff
@@ -904,11 +904,20 @@ def add_link_flag(i, f):
 
       link_flags.append((i, f))
 
+    # variable to check whether stdin contains code
+    pipe_stdin = None
+
     # find input files with a simple heuristic. we should really analyze
     # based on a full understanding of gcc params, right now we just assume that
     # what is left contains no more |-x OPT| things
     for i in range(len(newargs)):
       arg = newargs[i]
+
+      if arg == '-' and not sys.stdin.isatty():
+        pipe_stdin = sys.stdin.read()
+        input_files.append((i, arg))
+        continue
```

Comment:
Can you move this code down 1001 and make it part of large if/else chain there?  No need to `continue`, and you can remove the input arg via `newargs[i] = ''` in the same way that other args are handled.